pkg/system/system.go

```diff
@@ -384,15 +384,20 @@ func (sc *Controller) upgradeNydusDaemon(d *daemon.Daemon, c upgradeRequest, man
 		return err
 	}
 
-	su := manager.SupervisorSet.GetSupervisor(d.ID())
-	if err := su.SendStatesTimeout(time.Second * 10); err != nil {
-		return errors.Wrap(err, "Send states")
+	if d.Supervisor != nil {
```

Comment:
Maybe change to:

```
if d.Supervisor == nil {
  return errors.New("should set recover policy to failover to enable hot upgrade")
}
```

And put it at the beginning of the func.

Reply:
done.